src/components/HubSpotForm.tsx

```diff
@@ -0,0 +1,323 @@
+    }
+
+    /**
+     * If ChiliPiper is enabled, load the script and add an event
+     * listener to map the lead data to the ChiliPiper scheduler.
+     */
+    if (chiliPiper) {
+        await loadScriptElement('chilipiper', chiliPiperScript)
+
+        const chiliPiperDomain = 'sourcegraph'
+        const chiliPiperRouter = 'contact-sales'
+
+        window.addEventListener('message', event => {
+            const data = event.data as HubSpotEventProps
+
+            if (data.type === 'hsFormCallback' && data.eventName === 'onFormSubmit') {
+                const lead = data.data.reduce((object, item) => Object.assign(object, { [item.name]: item.value }), {})
+                const chiliPiperWindow = window.ChiliPiper
+
+                // Prevent schdulder from opening twice
+                const scheduler = document.querySelector('.chilipiper-popup')
+                if (scheduler) {
+                    return
+                }
+
+                chiliPiperWindow?.submit(chiliPiperDomain, chiliPiperRouter, {
+                    map: true,
+                    lead,
+                    closeOnOutside: true
+                })
+            }
+        })
+    }
+}
+
+/**
+ * This creates the HubSpot form with the configuration options.
+ * See: https://legacydocs.hubspot.com/docs/methods/forms/advanced_form_options
+ *
+ * @param CreateHubSpotFormProps - object props passed to createHubSpotForm
+ * @param CreateHubSpotFormProps.formId - the form's id
+ * @param CreateHubSpotFormProps.onFormReady - callback after form is built
+ * @param CreateHubSpotFormProps.onFormSubmitted - callback after data is sent
+ * @param CreateHubSpotFormProps.inlineMessage - form submission message
+ */
+function createHubSpotForm({ formId, onFormReady, onFormSubmitted, inlineMessage }: CreateHubSpotFormProps): void {
+    const script = getScriptElement('hubspot')
+
+    // When the HubSpot script is loaded, create the form with the config
+    script?.addEventListener('load', () => {
+        window.hbspt?.forms.create({
+            region: 'na1',
+            portalId: '2762526',
+            formId,
+            target: '#form-target',
+            onFormReady: (form: CreateHubSpotFormProps) => {
+                if (onFormReady) {
+                    onFormReady(form[0])
+                }
+            },
+            onFormSubmitted,
+            inlineMessage,
+        })
+    })
+}
+
+// This gets called when the HubSpot form is ready
+const onFormReady = (form: HTMLFormElement): void => {
+    /**
+     * This allows you to populate hidden form fields with values
+     *
+     * @param formField - the form field name
+     * @param value - the value to populate
+     */
+    const populateHiddenFormField = (formField: string, value: string): void => {
+        const input = form[0].querySelector(`input[name="${formField}"]`) as HTMLInputElement
+        if (input && input.value === '') {
```

Comment:
This is nitpitcky, but I prefer these statements to be written like `!input.value`. Feels more shorthand. `var === false` vs `!var`. Do you like it b/c it's more verbose?

Reply:
Ah, this was probably just an oversight on my part. I pushed up a fix in my latest commit. 🤓 Thanks Becca!